plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java

```diff
@@ -155,7 +150,11 @@ public void onFailure(Throwable e) {
         void executeScroll(final String scrollId) {
             try {
                 logger.trace("executing scroll request [{}]", scrollId);
-                scrollAction.execute(new SearchScrollRequest(scrollId).scroll(request.scroll()), new ActionListener<SearchResponse>() {
+                SearchScrollRequest searchScrollRequest = new SearchScrollRequest();
+                searchScrollRequest.copyContextAndHeadersFrom(request);
```

Comment:
can you instead add a new constructor to SearchScrollRequest that accepts the original request like we do elsewhere?